### ISSUE TITLE
triggerBuild API call acknowledges success, but project is never built

### ISSUE THREAD

**Issue by pquentin** [Shippable/support#1793]
I'm using the Shippable API to build all our projects every day. Unfortunately, one project added recently (55bf3993edd7f2c05291f1c4) never gets built by the API.

The `triggerBuild` call is done with `apiToken` and `projectId`, but without a `branchName` in order to use the default branch. It returns 200, and a `buildId` is even returned (see the list below), but the project does not actually get built. The only way to fix the issue is to use the web interface (with the same user that generated the `apiToken`).

Here's a list of `buildId`s for the past half-hour:

```
55d16dd38083110b0048eadd
55d16ca48083110b0048e60c
55d16b748083110b0048e561
55d16a458083110b0048dfb4
55d169168083110b0048df86
55d167e78083110b0048de41
55d166b88083110b0048dd54
55d165888083110b0048d1ab
```

Anything I could do to help get this bug fixed? Thanks!

**Comment by lekhab** [comment]
opened an internal issue to figure out what's going on.

**Comment by rageshkrishna** [comment]
@pquentin Would it be alright if we try to trigger some builds against your project to see what's happening here? Thanks!

**Comment by pquentin** [comment]
@rageshkrishna sure, go ahead. Thanks for looking into this!

**Comment by rageshkrishna** [comment]
@pquentin I tried trigger a few builds without specifying any branch and it seems to work now (it's kicking off a build on master). I'm not really sure why it was failing earlier. I suspect we were not able to synchronize the default branch name for the repository and it got populated later.

Can you please try this again and let me know if you're still running in to issues?

**Comment by pquentin** [comment]
@rageshkrishna Thank you for the investigation. It does seem to work now!

For this repository, we forgot to change the Bitbucket default branch from master to develop, so I don't think this setting was ever blank and populated later. Anyway, I changed Bitbucket default branch from master to develop, synced the project, and it seems to work now: I triggered a few builds from the script that exposed the issues and it did build develop.

:100:
